compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1197,6 +1227,9 @@ TR::Instruction *OMR::Power::MemoryReference::expandInstruction(TR::Instruction
    // a memory instruction.
    TR_ASSERT_FATAL(currentInstruction->getPrev(), "The first instruction cannot be a memory instruction");
 
+   if (self()->getLabel())
```

Comment:
Could you please elaborate why expansion is not needed?

Reply:
This is a temporary band-aid to allow tests to work correctly before this function has full support for prefixed instructions. While there's another check later to handle ignoring prefixed instructions in general, there's one extra restriction on references to labels that causes issues right now. Specifically, for simplicity in the relevant binary encoding functions, `MemoryReference`s with and without labels are treated differently in terms of their base register. Without a label, it's expected that a `NULL` base register should always be replaced with `gr0` so that we can unconditionally use the `fillFieldRA` helper. With a label, a very different code path is taken due to the need to handle potential relocations and that code path assumes that the base register will be `NULL`, so performing such a replacement would be undesirable.

I've got a slightly more elegant solution to this ready on my branch for adding correct handling of prefixed instructions in this method that renders this hack obsolete. You can see the WIP changes that clean this up at https://github.com/aviansie-ben/omr/tree/p10-ptoc-replacement and those are expected to be submitted within 2-4 weeks. Unfortunately, the changes there to make this function correctly handle this case require some other changes to go in alongside them that are still undergoing testing on a P10 simulator, so I can't include them in this PR right now.